token/cli/src/sort.rs

```diff
@@ -1,99 +1,116 @@
-                        let entry = mint_accounts.entry(mint);
-                        match entry {
-                            Entry::Occupied(_) => {
-                                entry.and_modify(|e| e.push(parsed_account));
-                            }
-                            Entry::Vacant(_) => {
-                                entry.or_insert_with(|| vec![parsed_account]);
-                            }
+                            .len(),
+                    );
+
+                    let cli_account = CliTokenAccount {
+                        address: address_str,
+                        program_id: program_id.to_string(),
+                        account: ui_token_account,
+                        is_associated,
+                    };
+
+                    let entry = cli_accounts.entry(btree_key);
+                    match entry {
+                        Entry::Occupied(_) => {
+                            entry.and_modify(|e| {
+                                if is_associated {
+                                    e.insert(0, cli_account)
+                                } else {
+                                    e.push(cli_account)
+                                }
+                            });
+                        }
+                        Entry::Vacant(_) => {
+                            entry.or_insert_with(|| vec![cli_account]);
                         }
                     }
-                    Ok(_) => unsupported_accounts.push(UnsupportedAccount {
-                        address,
-                        err: "Not a token account".to_string(),
-                    }),
-                    Err(err) => unsupported_accounts.push(UnsupportedAccount {
-                        address,
-                        err: format!("Account parse failure: {}", err),
-                    }),
                 }
+                Ok(_) => unsupported_accounts.push(UnsupportedAccount {
+                    address: address_str,
+                    err: "Not a token account".to_string(),
+                }),
+                Err(err) => unsupported_accounts.push(UnsupportedAccount {
+                    address: address_str,
+                    err: format!("Account parse failure: {}", err),
+                }),
             }
-        } else {
-            unsupported_accounts.push(UnsupportedAccount {
-                address,
-                err: "Unsupported account data format".to_string(),
-            });
         }
     }
-    for (_, array) in mint_accounts.iter_mut() {
-        array.sort_by(|a, b| b.is_associated.cmp(&a.is_associated));
-    }
-    (
-        mint_accounts,
+
+    Ok(CliTokenAccounts {
+        accounts: cli_accounts
+            .into_iter()
+            .map(|(_key, accounts_list)| accounts_list)
+            .collect(),
         unsupported_accounts,
         max_len_balance,
-        includes_aux,
-    )
+        aux_len: if includes_aux { 10 } else { 0 },
```

Comment:
nit: instead of `10`, could this be `" (Aux-{}*)".chars().count()` or some const formatting string imported from output?

Reply:
ill do you one better and get the number right whether theres above or below 9 aux accounts